src/node_http_parser.cc

```diff
@@ -472,6 +472,16 @@ class Parser : public AsyncWrap {
   static void Consume(const FunctionCallbackInfo<Value>& args) {
     Parser* parser;
     ASSIGN_OR_RETURN_UNWRAP(&parser, args.Holder());
+
+    // Argument coming from JS land with unchecked type.
+    // Make sure it is a valid external object.
+    if (!args[0]->IsExternal()) {
+      parser->env()->isolate()->ThrowException(Exception::TypeError(
+        String::NewFromUtf8(parser->env()->isolate(),
+        "Object conversion of argument failed.")));
+      return;
+    }
```

Comment:
```
out/Release/node[8983]: ../src/node_http_parser.cc:486:static void node::Parser::Consume(const FunctionCallbackInfo<v8::Value> &): Assertion `args[0]->IsExternal()' failed.
...

```
With the CHECK macro, the failure is a simple assertion as above (with callstack) and the message is not very intuitive and hence I went with a throw.


Regarding throw: made the changes are suggested.

Hope this is fine by you?